### PR TITLE
Rework org perms again

### PR DESCRIPTION
**Pull request by rowanseymour**
Reworked to make it easier to redirect regular users to a org-switcher page if they access a URL for an object that belongs to an org they also belong to (like what we do for staff).

### CODE REVIEW

File: temba/orgs/views/tests.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def test_has_permission(self):
         self.assertEqual(200, self.client.get(create_url).status_code)
 
         # staff still can't POST
-        self.assertLoginRedirect(self.client.post(create_url, {"name": "Sales"}))
+        self.assertEqual(403, self.client.post(create_url, {"name": "Sales"}).status_code)
```
Comment (rowanseymour) [inline temba/orgs/views/tests.py:51]: @ericnewcomer ok so this is changing again.. but I think this is safer. You POST as staff, the request just fails. No redirecting to login and then maybe back and wondering what happened with the original form submission.

File: temba/channels/tests.py [hunk 2]
```diff
@@ -972,8 +972,7 @@ def test_configuration(self):
         config_url = reverse("channels.channel_configuration", args=[self.ex_channel.uuid])
 
         # can't view configuration if not logged in
-        response = self.client.get(config_url)
-        self.assertLoginRedirect(response)
+        self.assertRequestDisallowed(config_url, [None, self.agent])
```
Comment (rowanseymour) [inline temba/channels/tests.py:975]: really gotta figure out a consistent behavior for when you hit an object specific URL for an object you don't have access to (but might by switching orgs). Here we're changing to a 404 from a login redirect.

File: temba/orgs/views/base.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def derive_queryset(self, **kwargs):
 
         # don't filter by org for staff users but let OrgObjPermsMixin provide a redirect
         if not self.request.user.is_staff:
-            qs = qs.filter(org=self.request.org)
+            qs = qs.filter(org__users=self.request.user)
```
Comment (rowanseymour) [inline temba/orgs/views/base.py:40]: if we were brave we could just not filter by org at all and trust `OrgObjPerms` to do its job

File: temba/orgs/views/mixins.py [hunk 4]
```diff
@@ -86,20 +77,30 @@ def get_object_org(self):
...
-            )
+        if self.request.org != org:
+            if request.user.is_staff:
+                # staff users are redirected to service page if org doesn't match
+                return HttpResponseRedirect(
+                    f"{reverse('staff.org_service')}?next={quote_plus(request.path)}&other_org={org.id}"
+                )
+            else:
+                # TODO implement view to let regular users switch orgs as well
+                return HttpResponseRedirect(reverse("orgs.org_choose"))
```
Comment (norkans7) [inline temba/orgs/views/mixins.py:103]: we can allow the choose view to accept a get param and use that for initial value of the form and the use can confirm to switch by a button(submitting the form)

Comment (rowanseymour) [reply temba/orgs/views/mixins.py:103]: we could but I think it's ok to have a different view for switching to a specific org

**Review by norkans7 — APPROVED** [review]

**Pull request merged**